client/settings/express-checkout-settings/payment-request-button-preview.js

```diff
@@ -101,87 +149,80 @@ const PaymentRequestButtonPreview = () => {
 			}
 			setIsLoading( false );
 		} );
-	}, [ stripe, setPaymentRequest, setIsLoading ] );
+	}, [ stripe, setPaymentRequest, setIsLoading, isStripeEceEnabled ] );
 
 	/**
 	 * If stripe is loading, then display nothing.
 	 * If stripe finished loading but payment request button failed to load (null), display info section.
 	 * If stripe finished loading and payment request button loads, display the button.
 	 */
 
-	return (
-		<>
-			{ ( isWooPayEnabled ||
-				( isPaymentRequestEnabled && paymentRequest ) ) && (
-				<div
-					className="payment-method-settings__preview"
-					data-theme={ theme }
-				>
-					{ isWooPayEnabled && (
-						<WoopayExpressCheckoutButton
-							isPreview={ true }
-							buttonSettings={ {
+	// No need to check `isStripeEceEnabled` since that's not what controls whether the express checkout
+	// buttons are displayed or not, that's always controlled by `isPaymentRequestEnabled`.
+	if ( ! isWooPayEnabled && ! isPaymentRequestEnabled ) {
+		return (
+			<InlineNotice icon status="info" isDismissible={ false }>
+				{ __(
+					'To preview the express checkout buttons, ' +
+						'activate at least one express checkout.',
+					'woocommerce-payments'
+				) }
+			</InlineNotice>
+		);
+	}
+
+	if ( isWooPayEnabled ) {
```

Comment:
I'm wondering about why we need to disable WooPay in order to test this? I guess we could handle the payment method separately and be able to preview both ECE and WooPay in their respective pages? Can you shed some light into this?

Reply:
That's what the previous implementation looked like, so I just copied that in my refactor 🙂 

I _think_ it has something to do with being able to reuse the component for both the WooPay and Express Checkout customization pages, but I'll admit I'm not entirely sure.

I just noticed that the WooPay preview actually looks really broken from the WooPay button settings right now, but it's like that in `develop` as well so I don't think it's a regression in this PR.

Reply:
>I just noticed that the WooPay preview actually looks really broken from the WooPay button settings right now, but it's like that in develop as well so I don't think it's a regression in this PR.

Turns out this was because I was messing with the settings in the DB, oops. You can ignore that part 😛 

> why we need to disable WooPay in order to test this?

You were right to call this out! It's supposed to show the WooPay button and express checkout buttons side-by-side, I just made a mistake during the refactor ☺️  Nice catch!

Reply:
Fixed in faa98c9816ee7a83217fce58306b58590487c9cd <3